### PR TITLE
PCHR-1533 fix layout on advanced search action subpages

### PR DESCRIPTION
**Pull request by danilo-compucorp**
This PR improves the layout for the "export contact" page. It was also used to do some code refactorings.
Please note that it also implements the "custom radio buttons and checkboxes" as per the invision layout. In order to customize these elements, the standard inputs are hidden, and a font-awesome icon will replace it. 
This is possible since the sibling _label_ (present immediately after the _input_) contains the _for_ attribute for the matching radio/checkbox id, and as this condition is not always present, a javascript file ( checkbox.js ) was added to try to add the missing _for_ attribute, or to make the standard radio/checkbox visible again when there's no sibling _label_.
In other words, when we have the following:
`<input type="radio" id="radio-id">
<label for="radio-id">Test</label>`
The radio button would be hidden ( `display: none` ), and a font-awesome icon would be added before the label, replacing the radio button.

**Example**:
radio

In order to avoid issues with markup inconsistencies, if the _input_ element lacks the _id_ attribute, the script would add a random id to it. 
For example: 
from this 
`<input type="radio">` 
to this 
`<input type="radio" id="1472062278795">`

If the _label_ element lacks the _for_ attribute, the script would add it, setting its value to the matching _input_ id.
For example: 
from this 
`<input type="radio" id="radio-id">
<label>Test</label>`
to this 
`<input type="radio" id="radio-id">
<label for="radio-id">Test</label>`

If a _label_ is not found immediately after the _input_, the script will make it visible, so we don't lose any feature. 
### Export contacts

**Before**:
before

**After**:
after2
### Add activity

**Before**
add-activity-before

**After**
add-activity
### Merge contacts

**Before**
merge-contacts-before

**After**
merge-contacts
### Print/Merge Document

**Before**
print-merge-before

**After**
print-merge
### Communication preferences

**Before**
communication-preferences-before

**After**
communication-preferences

### CODE REVIEW

**Comment by brunoqueiros** [comment]
Approved

File: org.civicrm.bootstrapcivicrm/scss/civicrm/common/_buttons.scss [hunk 1]
```diff
@@ -62,7 +62,7 @@ $button-border-radius: 2px;
     &.crm-search-form-block > .crm-submit-buttons,
     &.crm-form-block > .crm-submit-buttons {
       @extend %submit-buttons-civi;
-    }
+        }
```
Comment (brunoqueiros) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/common/_buttons.scss]: Wrong indentation

File: org.civicrm.bootstrapcivicrm/scss/civicrm/common/_buttons.scss [hunk 2]
```diff
@@ -83,6 +82,7 @@ $button-border-radius: 2px;
           color: $gray !important;
           border: solid 1px $gray-dark;
 
+
```
Comment (brunoqueiros) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/common/_buttons.scss]: Remove this empty line

File: org.civicrm.bootstrapcivicrm/scss/civicrm/common/_common.scss [hunk 3]
```diff
@@ -8,6 +8,7 @@
 @import 'modals';
 @import 'datepickers';
 @import 'radio-checkbox';
+
```
Comment (brunoqueiros) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/common/_common.scss]: Empty line

File: org.civicrm.bootstrapcivicrm/scss/civicrm/common/_forms.scss [hunk 4]
```diff
@@ -31,6 +31,7 @@ $control-height: 30px;
     }
   }
 
+
```
Comment (brunoqueiros) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/common/_forms.scss]: Empty line

File: org.civicrm.bootstrapcivicrm/scss/civicrm/common/_forms.scss [hunk 5]
```diff
@@ -162,11 +163,12 @@ $control-height: 30px;
       @extend %custom-form-control, %open-sans-font-family;
     }
 
-    // "Select2" Drop-downs
+    // 'Select2' Drop-downs
     .select2-container,
     .crm-form-select {
+
```
Comment (brunoqueiros) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/common/_forms.scss]: Empty line

File: org.civicrm.bootstrapcivicrm/scss/civicrm/common/_radio-checkbox.scss [hunk 6]
```diff
@@ -4,8 +4,10 @@
     display: none;
   }
   input[type=checkbox] + label[for],
-  input[type=radio] + label[for] {
+  input[type=radio] + label[for]{
+    vertical-align: middle;
     cursor: pointer;
+      margin-right: 15px !important;
```
Comment (brunoqueiros) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/common/_radio-checkbox.scss]: Wrong indentation

File: org.civicrm.bootstrapcivicrm/scss/civicrm/advanced-search/pages/_add-relationship.scss [hunk 7]
```diff
@@ -22,4 +24,24 @@
       color: #ffffff !important;
     }
   }
+
+  .form-layout-compressed {
+    margin: 15px !important;
+  }
+
+  .form-layout-compressed > tbody:nth-child(1) > tr:nth-child(4) {
```
Comment (brunoqueiros) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/advanced-search/pages/_add-relationship.scss]: Can You merge this selector with the selector on line 41?

File: org.civicrm.bootstrapcivicrm/scss/civicrm/advanced-search/pages/_add-relationship.scss [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+      background: $gray-dark !important;
+      color: #ffffff !important;
+    }
+  }
+
+  .form-layout-compressed {
+    margin: 15px !important;
+  }
+
+  .form-layout-compressed > tbody:nth-child(1) > tr:nth-child(4) {
```
Comment (brunoqueiros) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/advanced-search/pages/_add-relationship.scss]: Can You merge this selector with the selector on line 41?

**Comment by brunoqueiros** [comment]
Approved

File: org.civicrm.bootstrapcivicrm/scss/civicrm/common/_backgrounds.scss [hunk 9]
```diff
@@ -10,7 +10,9 @@
   }
 
   h1.page-title {
+
```
Comment (brunoqueiros) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/common/_backgrounds.scss]: Empty line

File: org.civicrm.bootstrapcivicrm/scss/civicrm/common/_buttons.scss [hunk 10]
```diff
@@ -4,11 +4,11 @@
 $button-border-radius: 2px;
 
 %btn-civi {
-  @extend .btn, %open-sans-font-family;
-  text-shadow: none;
-  text-transform: uppercase;
-  opacity: 1 !important;
-  font-size: 1em !important;
+    @extend .btn, %open-sans-font-family;
```
Comment (brunoqueiros) [inline org.civicrm.bootstrapcivicrm/scss/civicrm/common/_buttons.scss]: Wrong indentation

**Comment by davialexandre** [comment]
With shoreditch we don't need this anymore

**Pull request closed without merging**